src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
```

Comment:
Avoid .forEach:

Again, the same approach as used in the other two methods, you can solve for **sum** using a **.reduce**.
![image](https://github.com/pedro-ruas/mate-v2-js_average-ages/assets/54222238/33caa7d6-476a-4f4e-8d81-8ad8f5cfc5dc)
This approach is better for its readability and performance.